Application/EdFi.Ods.Features/ChangeQueries/Controllers/DeletesController.cs

```diff
@@ -76,13 +77,15 @@ public async Task<IActionResult> Get(string schema, string resource, [FromQuery]
             {
                 _logger.Debug("ChangeQueries is not enabled.");
 
-                return ControllerHelpers.NotFound("ChangeQueries is not enabled.");
+                return ControllerHelpers.FeatureDisabled(
+                    ChangeQueriesConstants.FeatureName,
+                    _logContextAccessor.GetCorrelationId());
```

Comment:
If I disable the `ChangeQueries` feature and call the deletes/keyChanges endpoint, I get a generic `System Error` because the Deletes/KeyChangesController can't be constructed since some of its dependencies aren't registered in the DI container because the feature is disabled.

`Autofac.Core.DependencyResolutionException: None of the constructors found with 'Autofac.Core.Activators.Reflection.DefaultConstructorFinder' on type 'EdFi.Ods.Features.ChangeQueries.Controllers.DeletesController' can be invoked with the available services and parameters:
Cannot resolve parameter 'EdFi.Ods.Features.ChangeQueries.Repositories.DeletedItems.IDeletedItemsResourceDataProvider deletedItemsResourceDataProvider' of constructor ...`

Reply:
I've added a new virtual method on the `ConditionalModule` class: `ApplyNullObjectPatternRegistrations`, which is called when the module returns `false` from `IsSelected`. It's virtual rather than abstract, so implementation by derived classes is optional. This allows us to implement the "Null Object Pattern" for dependencies that aren't used but need to be supplied for reasons like you've identified above.